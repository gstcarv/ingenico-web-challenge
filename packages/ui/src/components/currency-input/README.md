# ðŸ’° Currency Input Component

A specialized input component for currency values with automatic formatting, number validation, and customizable styling.

## ðŸ“¦ Installation

```bash
import { CurrencyInput } from '@ingenico-challenge/ui'
```

## ðŸš€ Usage

### Basic Currency Input

```tsx
import { CurrencyInput } from '@ingenico-challenge/ui'

function MyComponent() {
  const [amount, setAmount] = useState<number>()

  return (
    <CurrencyInput
      value={amount}
      onChange={setAmount}
      placeholder="Enter amount"
    />
  )
}
```

### Currency Input with Prefix/Suffix

```tsx
import { DollarSign } from 'lucide-react'

// With currency symbol prefix
<CurrencyInput
  value={amount}
  onChange={setAmount}
  prefix={<DollarSign size={20} />}
  placeholder="0.00"
/>

// With currency code suffix
<CurrencyInput
  value={amount}
  onChange={setAmount}
  suffix="USD"
  placeholder="0.00"
/>
```

### Different Sizes

```tsx
// Small input
<CurrencyInput size="sm" value={amount} onChange={setAmount} />

// Medium input (default)
<CurrencyInput size="md" value={amount} onChange={setAmount} />

// Large input
<CurrencyInput size="lg" value={amount} onChange={setAmount} />
```

### Controlled Component with Form Integration

```tsx
import { useForm } from 'react-hook-form'
import { CurrencyInput } from '@ingenico-challenge/ui'

function FormExample() {
  const { register, setValue, watch } = useForm()
  const amount = watch('amount')

  return (
    <form>
      <CurrencyInput
        value={amount}
        onChange={(value) => setValue('amount', value)}
        placeholder="Enter price"
      />
    </form>
  )
}
```

### Currency Input with Validation

```tsx
function ValidatedCurrencyInput() {
  const [amount, setAmount] = useState<number>()
  const [error, setError] = useState('')

  const handleChange = (value: number | undefined) => {
    setAmount(value)
    
    if (value && value > 10000) {
      setError('Amount cannot exceed $10,000')
    } else {
      setError('')
    }
  }

  return (
    <div>
      <CurrencyInput
        value={amount}
        onChange={handleChange}
        placeholder="Enter amount"
      />
      {error && <span className="text-red-500 text-sm">{error}</span>}
    </div>
  )
}
```

## ðŸ“‹ Props

| Prop | Type | Default | Description |
|------|------|---------|-------------|
| `value` | `number \| undefined` | - | Current currency value |
| `onChange` | `(value: number \| undefined) => void` | - | Callback when value changes |
| `size` | `'sm' \| 'md' \| 'lg'` | `'md'` | Input size variant |
| `placeholder` | `string` | `'0.00'` | Placeholder text |
| `prefix` | `ReactNode` | - | Content to display before input |
| `suffix` | `ReactNode` | - | Content to display after input |
| `className` | `string` | - | Additional CSS classes |
| `disabled` | `boolean` | `false` | Disables the input |
| `ref` | `Ref<HTMLInputElement>` | - | Ref to the input element |
| `...props` | `NumericFormatProps` | - | All NumericFormat props except overridden ones |

## ðŸŽ¨ Styling

The component uses Class Variance Authority (CVA) for consistent styling:

### Base Styling
- Full width and height
- Transparent background
- No border (handled by InputBase)
- Neutral text color
- Medium font weight
- Placeholder styling

### Size Variants
- **Small**: `text-sm px-3`
- **Medium**: `text-lg px-4` (default)
- **Large**: `text-2xl px-6`

### Focus States
- Focus ring removed (handled by InputBase)
- Custom focus styling through InputBase

## ðŸ”§ Features

### Automatic Formatting
- Thousand separators (commas)
- Decimal separator (period)
- Fixed 2 decimal places
- No negative values allowed

### Number Handling
- Converts string input to number
- Handles undefined values
- Maintains precision

### Accessibility
- Proper ARIA attributes
- Keyboard navigation
- Screen reader support
- Focus management

## ðŸ§ª Testing

```tsx
import { render, screen, fireEvent } from '@testing-library/react'
import { CurrencyInput } from './currency-input'

test('renders currency input with placeholder', () => {
  render(<CurrencyInput placeholder="Enter amount" />)
  expect(screen.getByTestId('currency-input')).toHaveAttribute('placeholder', 'Enter amount')
})

test('handles value changes', () => {
  const handleChange = jest.fn()
  render(<CurrencyInput onChange={handleChange} />)
  
  const input = screen.getByTestId('currency-input')
  fireEvent.change(input, { target: { value: '123.45' } })
  
  expect(handleChange).toHaveBeenCalledWith(123.45)
})

test('formats numbers correctly', () => {
  const handleChange = jest.fn()
  render(<CurrencyInput value={1234.56} onChange={handleChange} />)
  
  const input = screen.getByTestId('currency-input')
  expect(input).toHaveValue('1,234.56')
})
```

## ðŸ“š Storybook

View interactive examples and documentation in Storybook:

```bash
cd packages/ui
pnpm storybook
```

Navigate to the Currency Input component in the Storybook sidebar.

## ðŸŽ¯ Best Practices

1. **Always provide onChange handler** for controlled usage
2. **Use appropriate size** for your UI context
3. **Include currency symbols** for better UX
4. **Handle validation** for business rules
5. **Provide meaningful placeholders**
6. **Consider accessibility** with proper labels

## ðŸ”§ Customization

### Custom Currency Symbol

```tsx
import { Euro } from 'lucide-react'

<CurrencyInput
  value={amount}
  onChange={setAmount}
  prefix={<Euro size={20} className="text-green-600" />}
/>
```

### Custom Styling

```tsx
<CurrencyInput
  value={amount}
  onChange={setAmount}
  className="border-2 border-blue-500 rounded-lg"
  size="lg"
/>
```

### Integration with Form Libraries

```tsx
// With React Hook Form
const { register, setValue } = useForm()

<CurrencyInput
  onChange={(value) => setValue('amount', value)}
  {...register('amount')}
/>

// With Formik
<CurrencyInput
  value={values.amount}
  onChange={(value) => setFieldValue('amount', value)}
/>
```
