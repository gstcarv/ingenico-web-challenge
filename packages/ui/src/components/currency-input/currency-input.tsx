import { cva, type VariantProps } from "class-variance-authority";
import { ComponentProps, useCallback, useMemo, useState, useEffect } from "react";
import { cn } from "../../utils";
import { InputBase, InputBaseProps } from "../input-base";

const currencyInputVariants = cva(
    "w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0",
    {
        variants: {
            size: {
                sm: "text-sm px-3",
                md: "text-lg px-4",
                lg: "text-2xl px-6",
            },
        },
        defaultVariants: {
            size: "md",
        },
    },
);

// Currency symbol mapping
const CURRENCY_SYMBOLS: Record<string, string> = {
    USD: "$",
    EUR: "€",
    GBP: "£",
    JPY: "¥",
    CAD: "C$",
    AUD: "A$",
    CHF: "CHF",
    CNY: "¥",
    INR: "₹",
    BRL: "R$",
};

export type CurrencyInputProps = Omit<ComponentProps<"input">, "value" | "onChange" | "prefix"> &
    VariantProps<typeof currencyInputVariants> &
    InputBaseProps & {
        value?: number;
        onChange?: (value: number | undefined) => void;
        currency?: string;
        placeholder?: string;
        precision?: number;
        allowNegative?: boolean;
        ref?: React.RefObject<HTMLInputElement>;
    };

export const CurrencyInput = ({
    ref,
    className,
    size,
    value,
    onChange,
    placeholder = "0.00",
    precision = 2,
    allowNegative = false,
    prefix,
    suffix,
    currency,
    ...props
}: CurrencyInputProps) => {
    // Determine the prefix to use
    const inputPrefix = useMemo(() => {
        if (prefix !== undefined) {
            return prefix;
        }
        if (currency) {
            return CURRENCY_SYMBOLS[currency] || currency;
        }
        return CURRENCY_SYMBOLS.USD; // Default to USD symbol
    }, [prefix, currency]);

    // Internal state for the input value
    const [inputValue, setInputValue] = useState<string>("");

    // Update internal state when value prop changes
    useEffect(() => {
        if (value === undefined || value === null) {
            setInputValue("");
        } else {
            // Format the value for display
            const formatted = formatValue(value, precision);
            setInputValue(formatted);
        }
    }, [value, precision]);

    // Format value with commas and decimal places
    const formatValue = (num: number, decimalPlaces: number): string => {
        const rounded = Math.round(num * Math.pow(10, decimalPlaces)) / Math.pow(10, decimalPlaces);
        const parts = rounded.toString().split(".");
        parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ",");

        if (decimalPlaces === 0) {
            return parts[0];
        }

        if (parts.length === 1) {
            return parts[0] + "." + "0".repeat(decimalPlaces);
        }

        return parts[0] + "." + parts[1].padEnd(decimalPlaces, "0");
    };

    // Parse input value to number
    const parseValue = (input: string): number | undefined => {
        if (!input || input.trim() === "") {
            return undefined;
        }

        // Remove commas and spaces
        const cleanInput = input.replace(/[, ]/g, "");

        // Check if it's a valid number
        const parsed = parseFloat(cleanInput);
        if (isNaN(parsed)) {
            return undefined;
        }

        return parsed;
    };

    // Validate input string
    const isValidInput = (input: string): boolean => {
        if (!input || input.trim() === "") {
            return true;
        }

        // Allow only numbers, commas, spaces, dots, and optionally minus
        const validChars = allowNegative ? /^[0-9.,\-\s]*$/ : /^[0-9.,\s]*$/;
        if (!validChars.test(input)) {
            return false;
        }

        // Check for multiple decimal points
        const decimalCount = (input.match(/\./g) || []).length;
        if (decimalCount > 1) {
            return false;
        }

        // Check for negative values when not allowed
        if (!allowNegative && input.includes("-")) {
            return false;
        }

        return true;
    };

    const handleChange = useCallback(
        (e: React.ChangeEvent<HTMLInputElement>) => {
            const newValue = e.target.value;

            // Handle empty input
            if (!newValue || newValue.trim() === "") {
                setInputValue("");
                onChange?.(undefined);
                return;
            }

            // Only call onChange if the input is valid
            if (isValidInput(newValue)) {
                const parsed = parseValue(newValue);
                if (parsed !== undefined) {
                    // Format the value immediately for display
                    const formatted = formatValue(parsed, precision);
                    setInputValue(formatted);
                    onChange?.(parsed);
                } else {
                    setInputValue(newValue);
                }
            } else {
                setInputValue(newValue);
            }
        },
        [onChange, allowNegative, precision],
    );

    const handleBlur = useCallback(
        (e: React.FocusEvent<HTMLInputElement>) => {
            const currentValue = e.target.value;

            if (!isValidInput(currentValue)) {
                // Clear invalid input
                setInputValue("");
                onChange?.(undefined);
                return;
            }

            const parsed = parseValue(currentValue);
            if (parsed !== undefined) {
                // Format the value on blur
                const formatted = formatValue(parsed, precision);
                setInputValue(formatted);
                onChange?.(parsed);
            }
        },
        [onChange, precision],
    );

    return (
        <InputBase
            className={cn(
                "focus-within:ring-2 focus-within:ring-brand-primary focus-within:border-brand-primary",
                className,
            )}
            prefix={inputPrefix}
            suffix={suffix}
        >
            <input
                ref={ref}
                type="text"
                inputMode="decimal"
                value={inputValue}
                onChange={handleChange}
                onBlur={handleBlur}
                placeholder={placeholder}
                className={cn(currencyInputVariants({ size }), inputPrefix && "pl-12", suffix && "pr-12")}
                {...props}
            />
        </InputBase>
    );
};
