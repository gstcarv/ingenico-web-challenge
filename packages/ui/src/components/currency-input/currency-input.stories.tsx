import type { Meta, StoryObj } from "@storybook/react-vite";
import { CurrencyInput } from "./currency-input";
import { DollarSign } from "lucide-react";

const meta: Meta<typeof CurrencyInput> = {
    title: "Components/CurrencyInput",
    component: CurrencyInput,
    parameters: {
        layout: "centered",
    },
    tags: ["autodocs"],
    argTypes: {
        disabled: {
            control: "boolean",
        },
        currency: {
            control: "select",
            options: ["USD", "EUR", "GBP", "JPY", "CAD", "AUD", "CHF", "CNY", "INR", "BRL"],
        },
        size: {
            control: "select",
            options: ["sm", "md", "lg"],
        },
        precision: {
            control: "number",
            min: 0,
            max: 4,
        },
        allowNegative: {
            control: "boolean",
        },
    },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        placeholder: "Insert amount",
        decimalScale: 2,
        allowDecimals: true,
        fixedDecimalLength: 2,
        decimalSeparator: ".",
        groupSeparator: ",",
        suffix: "USD",
        prefix: "$",
    },
};

export const EuroCurrency: Story = {
    args: {
        currency: "EUR",
        placeholder: "0.00",
    },
};

export const BrazilianReal: Story = {
    args: {
        currency: "BRL",
        placeholder: "0.00",
    },
};

export const WithIconPrefix: Story = {
    render: () => <CurrencyInput prefix={<DollarSign size={16} />} value={1234.56} placeholder="0.00" />,
};

export const Disabled: Story = {
    args: {
        disabled: true,
        value: 1234.56,
        placeholder: "0.00",
    },
};

export const WithLabel: Story = {
    render: () => (
        <div className="space-y-2">
            <label htmlFor="amount" className="text-sm font-medium text-neutral-700">
                Amount
            </label>
            <CurrencyInput id="amount" name="amount" value={1234.56} placeholder="0.00" />
        </div>
    ),
};
