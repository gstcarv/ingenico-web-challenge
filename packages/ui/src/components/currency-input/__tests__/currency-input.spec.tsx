import { render, screen, fireEvent } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'
import { CurrencyInput } from '../currency-input'

describe('CurrencyInput', () => {
  it('renders input element with correct attributes', () => {
    render(<CurrencyInput data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    expect(inputElement).toBeInTheDocument()
    expect(inputElement.tagName).toBe('INPUT')
    expect(inputElement).toHaveAttribute('type', 'text')
    expect(inputElement).toHaveAttribute('inputMode', 'decimal')
  })

  it('applies custom className', () => {
    const { container } = render(
      <CurrencyInput className="custom-class" data-testid="currency-input" />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders default USD currency symbol', () => {
    render(<CurrencyInput data-testid="currency-input" />)

    expect(screen.getByText('$')).toBeInTheDocument()
  })

  it('renders custom currency symbol', () => {
    render(<CurrencyInput currency="EUR" data-testid="currency-input" />)

    expect(screen.getByText('â‚¬')).toBeInTheDocument()
  })

  it('renders custom prefix when provided', () => {
    render(<CurrencyInput prefix="R$" data-testid="currency-input" />)

    expect(screen.getByText('R$')).toBeInTheDocument()
  })

  it('renders React node prefix', () => {
    const TestIcon = () => <span data-testid="icon">ğŸ’°</span>

    render(<CurrencyInput prefix={<TestIcon />} data-testid="currency-input" />)

    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('renders suffix when provided', () => {
    render(<CurrencyInput suffix="USD" data-testid="currency-input" />)

    expect(screen.getByText('USD')).toBeInTheDocument()
  })

  it('handles value prop correctly', () => {
    const handleChange = vi.fn()

    render(
      <CurrencyInput value={1234.56} onChange={handleChange} data-testid="currency-input" />
    )

    const inputElement = screen.getByTestId('currency-input')
    expect(inputElement).toHaveValue('1,234.56')
  })

  it('handles empty value', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput value={undefined} onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    expect(inputElement).toHaveValue('')
  })

  it('formats input value on change', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    fireEvent.change(inputElement, { target: { value: '1234.56' } })

    expect(handleChange).toHaveBeenCalledWith(1234.56)
    expect(inputElement).toHaveValue('1,234.56')
  })

  it('handles empty input', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    // First set a value, then clear it
    fireEvent.change(inputElement, { target: { value: '123' } })
    fireEvent.change(inputElement, { target: { value: '' } })

    expect(handleChange).toHaveBeenCalledWith(123)
    expect(handleChange).toHaveBeenCalledWith(undefined)
    expect(inputElement).toHaveValue('')
  })

  it('handles invalid input gracefully', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    fireEvent.change(inputElement, { target: { value: 'abc' } })

    expect(handleChange).not.toHaveBeenCalled()
    expect(inputElement).toHaveValue('abc')
  })

  it('handles multiple decimal points', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    fireEvent.change(inputElement, { target: { value: '123.45.67' } })

    expect(handleChange).not.toHaveBeenCalled()
    expect(inputElement).toHaveValue('123.45.67')
  })

  it('formats value on blur', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    fireEvent.change(inputElement, { target: { value: '1234' } })
    fireEvent.blur(inputElement)

    expect(inputElement).toHaveValue('1,234.00')
  })

  it('clears invalid value on blur', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    fireEvent.change(inputElement, { target: { value: 'abc' } })
    fireEvent.blur(inputElement)

    expect(inputElement).toHaveValue('')
  })

  it('handles different precision values', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput precision={0} onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    fireEvent.change(inputElement, { target: { value: '1234.56' } })

    expect(handleChange).toHaveBeenCalledWith(1234.56)
    expect(inputElement).toHaveValue('1,235')
  })

  it('allows negative values when allowNegative is true', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput allowNegative onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    fireEvent.change(inputElement, { target: { value: '-1234.56' } })

    expect(handleChange).toHaveBeenCalledWith(-1234.56)
    expect(inputElement).toHaveValue('-1,234.56')
  })

  it('prevents negative values when allowNegative is false', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput allowNegative={false} onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    fireEvent.change(inputElement, { target: { value: '-1234.56' } })

    expect(handleChange).not.toHaveBeenCalled()
    expect(inputElement).toHaveValue('-1234.56')
  })

  it('uses custom placeholder', () => {
    render(<CurrencyInput placeholder="Enter amount" data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    expect(inputElement).toHaveAttribute('placeholder', 'Enter amount')
  })

  it('uses default placeholder', () => {
    render(<CurrencyInput data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    expect(inputElement).toHaveAttribute('placeholder', '0.00')
  })

  it('handles different size variants', () => {
    const { container } = render(
      <CurrencyInput size="lg" data-testid="currency-input" />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('forwards all HTML input attributes', () => {
    render(
      <CurrencyInput
        data-testid="currency-input"
        name="amount"
        id="amount-input"
        required
        aria-label="Enter amount"
        disabled
      />
    )

    const inputElement = screen.getByTestId('currency-input')
    expect(inputElement).toHaveAttribute('name', 'amount')
    expect(inputElement).toHaveAttribute('id', 'amount-input')
    expect(inputElement).toHaveAttribute('required')
    expect(inputElement).toHaveAttribute('aria-label', 'Enter amount')
    expect(inputElement).toBeDisabled()
  })

  it('handles ref forwarding', () => {
    const ref = vi.fn()

    render(<CurrencyInput ref={ref} data-testid="currency-input" />)

    expect(ref).toHaveBeenCalled()
  })

  it('supports different currency codes', () => {
    const currencies = ['EUR', 'GBP', 'JPY', 'CAD', 'AUD', 'CHF', 'CNY', 'INR', 'BRL']
    const symbols = ['â‚¬', 'Â£', 'Â¥', 'C$', 'A$', 'CHF', 'Â¥', 'â‚¹', 'R$']

    currencies.forEach((currency, index) => {
      const { unmount } = render(
        <CurrencyInput currency={currency} data-testid="currency-input" />
      )

      expect(screen.getByText(symbols[index])).toBeInTheDocument()
      unmount()
    })
  })

  it('falls back to currency code when symbol not found', () => {
    render(<CurrencyInput currency="XYZ" data-testid="currency-input" />)

    expect(screen.getByText('XYZ')).toBeInTheDocument()
  })
})

describe('CurrencyInput Integration', () => {
  it('works with form elements', () => {
    const handleSubmit = vi.fn((e) => e.preventDefault())

    render(
      <form onSubmit={handleSubmit}>
        <CurrencyInput name="amount" data-testid="currency-input" />
        <button type="submit">Submit</button>
      </form>
    )

    const inputElement = screen.getByTestId('currency-input')
    const submitButton = screen.getByText('Submit')

    fireEvent.change(inputElement, { target: { value: '1234.56' } })
    fireEvent.click(submitButton)

    expect(handleSubmit).toHaveBeenCalledTimes(1)
  })

  it('handles keyboard navigation', () => {
    const handleChange = vi.fn()

    render(<CurrencyInput onChange={handleChange} data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')

    // Focus the input
    inputElement.focus()
    expect(inputElement).toHaveFocus()

    // Simulate keyboard input
    fireEvent.change(inputElement, { target: { value: '1234.56' } })

    expect(handleChange).toHaveBeenCalledWith(1234.56)
  })
})

describe('CurrencyInput Accessibility', () => {
  it('supports aria-label', () => {
    render(<CurrencyInput aria-label="Enter amount" data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    expect(inputElement).toHaveAttribute('aria-label', 'Enter amount')
  })

  it('supports aria-describedby', () => {
    render(<CurrencyInput aria-describedby="description" data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    expect(inputElement).toHaveAttribute('aria-describedby', 'description')
  })

  it('supports role attribute', () => {
    render(<CurrencyInput role="spinbutton" data-testid="currency-input" />)

    const inputElement = screen.getByTestId('currency-input')
    expect(inputElement).toHaveAttribute('role', 'spinbutton')
  })
})
