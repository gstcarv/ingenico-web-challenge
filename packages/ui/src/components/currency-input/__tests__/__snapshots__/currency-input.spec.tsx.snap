// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CurrencyInput > applies correct base classes to container 1`] = `
<div
  class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium"
>
  <input
    class="w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0 text-lg px-4"
    data-testid="currency-input"
    inputmode="numeric"
    placeholder="0.00"
    type="text"
    value=""
  />
</div>
`;

exports[`CurrencyInput > applies correct classes to input element 1`] = `
<div
  class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium"
>
  <input
    class="w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0 text-lg px-4"
    data-testid="currency-input"
    inputmode="numeric"
    placeholder="0.00"
    type="text"
    value=""
  />
</div>
`;

exports[`CurrencyInput > applies correct padding when prefix is present 1`] = `
<div
  class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium"
>
  <div
    class="absolute left-4 top-1/2 transform -translate-y-1/2 text-neutral-500"
  >
    R$
  </div>
  <input
    class="w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0 text-lg px-4 pl-12"
    data-testid="currency-input"
    inputmode="numeric"
    placeholder="0.00"
    type="text"
    value=""
  />
</div>
`;

exports[`CurrencyInput > applies custom className 1`] = `
<div
  class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium custom-class"
>
  <input
    class="w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0 text-lg px-4"
    data-testid="currency-input"
    inputmode="numeric"
    placeholder="0.00"
    type="text"
    value=""
  />
</div>
`;

exports[`CurrencyInput > applies default padding when no prefix is present 1`] = `
<div
  class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium"
>
  <input
    class="w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0 text-lg px-4"
    data-testid="currency-input"
    inputmode="numeric"
    placeholder="0.00"
    type="text"
    value=""
  />
</div>
`;

exports[`CurrencyInput > applies different sizes correctly 1`] = `
<div
  class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium"
>
  <input
    class="w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0 text-lg px-4"
    data-testid="currency-input"
    inputmode="numeric"
    placeholder="0.00"
    type="text"
    value=""
  />
</div>
`;

exports[`CurrencyInput > applies different sizes correctly 2`] = `
<div
  class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium"
>
  <input
    class="w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0 text-lg px-4"
    data-testid="currency-input"
    inputmode="numeric"
    placeholder="0.00"
    type="text"
    value=""
  />
</div>
`;

exports[`CurrencyInput Variants > applies large size variant 1`] = `
<div
  class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium"
>
  <input
    class="w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0 text-lg px-4"
    data-testid="currency-input"
    inputmode="numeric"
    placeholder="0.00"
    type="text"
    value=""
  />
</div>
`;

exports[`CurrencyInput Variants > applies medium size variant (default) 1`] = `
<div
  class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium"
>
  <input
    class="w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0 text-lg px-4"
    data-testid="currency-input"
    inputmode="numeric"
    placeholder="0.00"
    type="text"
    value=""
  />
</div>
`;

exports[`CurrencyInput Variants > applies small size variant 1`] = `
<div
  class="input-base h-16 w-70 bg-white border border-neutral-200 rounded-lg shadow-lg relative text-neutral-500 font-medium"
>
  <input
    class="w-full h-full bg-transparent border-none outline-none text-neutral-900 font-medium placeholder:text-neutral-400 focus-visible:ring-0 text-lg px-4"
    data-testid="currency-input"
    inputmode="numeric"
    placeholder="0.00"
    type="text"
    value=""
  />
</div>
`;
